frontend/public/components/impersonate-notifier.jsx

```diff
@@ -1,24 +1,38 @@
-import * as _ from 'lodash-es';
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { useTranslation, Trans } from 'react-i18next';
+import { Button } from '@patternfly/react-core';
 
 import * as UIActions from '../actions/ui';
+import { modelFor } from '../module/k8s';
 
 export const ImpersonateNotifier = connect(({ UI }) => ({ impersonate: UI.get('impersonate') }), {
   stopImpersonate: UIActions.stopImpersonate,
 })(({ stopImpersonate, impersonate }) => {
+  const { t } = useTranslation();
   if (!impersonate) {
     return null;
   }
+  const kindTranslated = modelFor(impersonate.kind)?.labelKey
+    ? t(modelFor(impersonate.kind).labelKey)
+    : impersonate.kind;
   return (
     <div className="co-global-notification">
       <div className="co-global-notification__content">
         <p className="co-global-notification__text">
-          <span className="text-uppercase co-global-notification__impersonate-kind">{`Impersonating ${impersonate.kind}`}</span>{' '}
-          You are impersonating{' '}
-          <span className="co-global-notification__impersonate-name">{impersonate.name}</span>. You
-          are viewing all resources and roles this {_.toLower(impersonate.kind)} can access.{' '}
-          <a onClick={stopImpersonate}>Stop Impersonation</a>
+          <span className="text-uppercase co-global-notification__impersonate-kind">
+            {t('public~Impersonating {{kind}}', {
+              kind: kindTranslated,
+            })}
+          </span>{' '}
+          <Trans t={t} ns="public">
+            You are impersonating{' '}
+            <span className="co-global-notification__impersonate-name">{impersonate.name}</span>.
+            You are viewing all resources and roles this {kindTranslated} can access.
```

Comment:
Do we need double braces here?
```suggestion
            <span className="co-global-notification__impersonate-name">{{impersonate.name}}</span>.
            You are viewing all resources and roles this {{kindTranslated}} can access.
```

Reply:
@spadgett I'm not sure...I've seen both in the code... @rebeccaalpert are a double set of braces required within the <Trans component for vars?

Reply:
I'm not sure if it's required, but the docs do use double brackets for what it's worth.

Reply:
Good to know, I'll double up!